source/blog/2016-07-25-ember-2-7-and-2-8-beta-released.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Ember.js 2.7 and 2.8 Beta Released
+author: Dan Gebhardt
+tags: Releases
+---
+
+Ember.js 2.7, a minor version release of Ember with backwards compatible
+changes, is released today.
+
+Ember.js 2.8 beta is also being released today. This branch will be released as
+stable in roughly six weeks and will then go on to be the next LTS release
+roughly six weeks after that.
+
+### Changes in Ember.js 2.7
+
+Ember 2.7 introduces one new backward compatible API:
+
+#### `Ember.computed.uniqBy`
+
+In general, the core team is hesitant about adding any new computed macros
+or array prototype extensions to the framework. Instead, we encourage developers
+to adopt an addon such as [ember-cpm](https://github.com/cibernox/ember-cpm)
+if they desire an expanded list of macros.
+
+However contributor [@seanjohnson08](https://github.com/seanjohnson08) correctly
+pointed out there is a logical gap in the APIs Ember provides today. Although
+each of `find`, `map`, `filter`, and `reject` have a comparable `xxxBy` method
+(for example `findBy`) no comparable method exists for `uniq`.
+
+Ember 2.7 adds `uniqBy` as an computed property macro. For example:
+
+```js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  accidents: Ember.computed(function() {
+    return [{
+      description: 'Flipped over',
+      car: { id: 1, make: 'Saturn' }
+    },{
+      description: 'Aliens',
+      car: { id: 2, make: 'Toyota' }
+    },{
+      description: 'Road rage',
+      car: { id: 1, make: 'Saturn' }
+    }];
+  }),
+  carsInAccidents: Ember.computed.mapBy('accidents', 'car'),
+  uniqueCarsInAccidents: Ember.computed.uniqBy('carsInAccidents', 'id')
+});
+```
+
+`uniqBy` is also added as an array method. For example:
+
+```js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  accidents: Ember.computed(function() {
+    return [{
+      description: 'Flipped over',
+      car: { id: 1, make: 'Saturn' }
+    },{
+      description: 'Aliens',
+      car: { id: 2, make: 'Toyota' }
+    },{
+      description: 'Road rage',
+      car: { id: 1, make: 'Saturn' }
+    }];
+  }),
+  uniqueCarsInAccidents: Ember.computed('accidents.@each.car', function() {
+    return this.get('accidents').mapBy('car').uniqBy('id');
+  })
+});
+```
+
+For more information on these methods, see the [API docs for the computed
+macro](http://emberjs.com/api/classes/Ember.computed.html#method_uniqBy), [API
+docs for the array
+method](http://emberjs.com/api/classes/Ember.Array.html#method_uniqBy), and [PR
+#12875](https://github.com/emberjs/ember.js/pull/12875) introducing these
+changes.
+
+#### Notable Deprecations
+
+The following deprecations are scheduled for release with Ember 2.7 and will be
+removed soon:
```

Comment:
Maybe ~~will be removed soon~~, because the second deprecation is for Ember 3.0 which is probably still a while away